File: src/com/example/javabutton/JavaPlayer.java
package com.example.javabutton;

import android.content.Context;
import android.net.Uri;

public class JavaPlayer {
	private JavaPlayerThread jThread;
	public JavaPlayer(Uri javaUri) {
		jThread=new JavaPlayerThread(javaUri);
		jThread.start();
	}
	public void stop() {
		jThread.enqueueStop();
	}
	public void play(Context ctx) {
		jThread.enqueueChangeContext(ctx);
		jThread.enqueuePlay();
	}
}

File: app/src/main/java/com/example/javabutton/JavaPlayer.kt
package com.example.javabutton

import android.content.Context
import android.net.Uri

class JavaPlayer(javaUri: Uri) {
    private val jThread: JavaPlayerThread

    init {
        jThread = JavaPlayerThread(javaUri)
        jThread.start()
    }

    fun play() {
        jThread.enqueuePlay()
    }

    fun stop() {
        jThread.enqueueStop()
    }

    fun setContext(ctx: Context) {
        jThread.enqueueChangeContext(ctx)
    }

    fun play(ctx: Context) {
        setContext(ctx)
        play()
    }
}
